src/types.ts

```diff
@@ -1,12 +1,11 @@
-/**
- * Retrieve the position of the current nodeView
- */
-type GetPosition = (() => number) | boolean;
+import { ProsemirrorAttributes } from 'remirror';
 
 export type EntityComponentProps = {
-	id: string;
-	name: string;
-	getPosition: GetPosition;
+	// The entity data of this component
+	entity: EntityAttrs;
+	updateAttributes: (attrs: ProsemirrorAttributes<object>) => void;
```

Comment:
I found the API a bit confusing:
- Wouldn't `updateEntity` be more logical here?
- Why can `updateAttributes` create new entities? I'd consider another function. If you really don't want this: maybe add at least a comment how to use the function to create new entities (and where they would be placed).